Game-Inventory.API/Controllers/CompaniesController.cs

```diff
@@ -8,43 +10,190 @@ namespace Game_Inventory.API.Controllers;
 [Route("[controller]")]
 public class CompaniesController : ControllerBase
 {
+
+    private readonly GameStopContext _context;
+
+    public CompaniesController(GameStopContext context)
+    {
+        _context = context;
+    }
+
     [HttpGet]
     [Route("/companies")]
     [SwaggerOperation(Tags = ["Companies"])]
-    public IActionResult GetCompanies()
+    public async Task<IActionResult> GetCompanies()
     {
-        throw new NotImplementedException();
+
+        try
+        {
+            var companies = await _context
+                .Companies
+                .ToListAsync();
+
+            return Ok(companies);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"An error ocurred: {ex.Message}");
+        }
     }
 
     [HttpGet]
     [Route("/companies/{id}")]
     [SwaggerOperation(Tags = ["Companies"])]
-    public IActionResult GetCompaniesById([FromRoute] int id)
+    public  async Task<IActionResult> GetCompaniesById([FromRoute] int id)
     {
-        throw new NotImplementedException();
+        var company = await _context
+            .Companies
+            .FindAsync(id);  
+
+        if(company is null)
+        {
+            var problemDetails = new ProblemDetails()
+            {
+                Status = 404,
```

Comment:
No es necesario mandarle el código aqui si abajo le estas devolviendo el actionresult not found